src/main/java/com/pingcap/tikv/util/ScalarFuncInfer.java

```diff
@@ -0,0 +1,85 @@
+package com.pingcap.tikv.util;
+
+import com.pingcap.tidb.tipb.ScalarFuncSig;
+import com.pingcap.tikv.exception.TypeException;
+import com.pingcap.tikv.types.*;
+
+import java.util.Objects;
+
+/**
+ * The ScalarFunction Signature inferrer.
+ * <p>
+ * Used to infer a target signature for the given DataType
+ */
+public class ScalarFuncInfer {
+  /**
+   * Infer scalar function signature.
+   * You should provide candidates for the
+   * inferrer to choose.
+   *
+   * @param dataType     the data type
+   * @param intSig       the int sig
+   * @param decimalSig   the decimal sig
+   * @param realSig      the real sig
+   * @param durationType the duration type
+   * @param timeType     the time type
+   * @return the scalar func sig
+   */
+  public static ScalarFuncSig infer(DataType dataType,
```

Comment:
This is quite confusing design here. It likely can be convert to a map lookup.

Reply:
Agree. Map lookup is much more faster.

Reply:
Thanks for the advice, please check the `Map` lookup plan in the newest commit.